src/core/fallcurve/internal/vt_fc_signature_compute.c

```diff
@@ -95,12 +97,12 @@ VT_UINT fc_signature_compute(
     // Find datapoint which reaches 37% of the starting value
     VT_INT index_37 = fc_signature_calculate_37index(raw_signature, sample_length);
     // fingerprint_length of this new fingerprint
-    if (index_37 > 0)
+    if (index_37 < sample_length)
     {
         sample_length = index_37 + 1;
     }
     // Calculate FallTime
-    falltime_computed = (VT_ULONG)sample_length * sampling_interval_us;
+    falltime_computed = (VT_ULONG)index_37 * sampling_interval_us;
```

Comment:
is this a bug fix? Seems to be a change in logic.

Reply:
Yes this is a change in logic for bug fix, explained in the reply to the next comment. I realised that this is not the right way to do it in terms of commits and PRs after I pushed it